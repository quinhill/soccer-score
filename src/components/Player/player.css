.container-container {
  display: flex;
  justify-content: center;
}

.player-container {
  display: grid;
  width: 500px;
  grid-template-columns: 50% 50%;
  background-color: #b5b5b7;
  border-radius: 5px;
  box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.6);
}

.player-name {
  color: #252839;
  margin: 10px;
}

.player-image {
  margin-bottom: 10px;
  box-shadow: 2px 2px 10px 0px rgba(0, 0, 0, 0.6);
}

.other-info-container {
  text-align: start;
}

.other-info {
  color: #252839;
}