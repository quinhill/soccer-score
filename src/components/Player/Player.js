import React from 'react';
import './player.css';
import { connect } from 'react-redux';

export const Player = (props) => {

  console.log(props)
  return (
    <div>

    </div>
  )
}

export const mapStateToProps = state => ({
  player: state.player
});

export default connect(mapStateToProps)(connect);