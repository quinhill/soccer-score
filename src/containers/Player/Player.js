import React, { Component } from 'react';
import './player.css';
import { connect } from 'react-redux';

export class Player extends Component {

  render() {
    return (
      <div className="container-container">
        <div className="player-container">
          <div className="name-image-container">
            <h1 className="player-name">{this.props.player.name}</h1>
            <img className="player-image" src={this.props.player.image} />
          </div>
          <div className="other-info-container">
            <p className="other-info"><span>Nationality:</span>{this.props.player.nationality}</p>
            <p className="other-info"><span>Date of Birth:</span>{this.props.player.birthdate}</p>
            <p className="other-info"><span>Birth Place:</span>{this.props.player.birthplace}</p>
            <p className="other-info"><span>Club:</span>{this.props.player.team}</p>
            <p className="other-info"><span>Position:</span>{this.props.player.position}</p>
          </div>
        </div>
      </div>
    )
  }
}

export const mapStateToProps = state => ({
  player: state.player
});

export default connect(mapStateToProps)(Player);