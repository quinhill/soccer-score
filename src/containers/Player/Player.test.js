import React from 'react';
import { shallow } from 'enzyme';
import { Player } from './Player';
import { mapStateToProps } from './Player';

describe('Player', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Player 
      player={{}}
    />)

    expect(wrapper).toMatchSnapshot()
  })

  describe('mapPropsToState', () => {
    it('should return a props object containing a player object', () => {
      const mockState = {
        liveScores: [{}, {}],
        player: {},
        team: { id: 'teamId' }
      }
      const expected = {
        player: {}
      }

      const mappedProps = mapStateToProps(mockState)

      expect(mappedProps).toEqual(expected);
    })
  })
})