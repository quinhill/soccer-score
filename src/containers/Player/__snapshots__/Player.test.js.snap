// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Player
    player={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="player-container"
>
        <div
                className="name-image-container"
        >
                <h1
                        className="player-name"
                />
                <img
                        className="player-image"
                        src={undefined}
                />
        </div>
        <div
                className="other-info-container"
        >
                <p
                        className="other-info"
                >
                        <span>
                                Nationality:
                        </span>
                </p>
                <p
                        className="other-info"
                >
                        <span>
                                Date of Birth:
                        </span>
                </p>
                <p
                        className="other-info"
                >
                        <span>
                                Birth Place:
                        </span>
                </p>
                <p
                        className="other-info"
                >
                        <span>
                                Club:
                        </span>
                </p>
                <p
                        className="other-info"
                >
                        <span>
                                Position:
                        </span>
                </p>
        </div>
</div>,
      "className": "container-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="name-image-container"
>
            <h1
                        className="player-name"
            />
            <img
                        className="player-image"
                        src={undefined}
            />
</div>,
          <div
            className="other-info-container"
>
            <p
                        className="other-info"
            >
                        <span>
                                    Nationality:
                        </span>
            </p>
            <p
                        className="other-info"
            >
                        <span>
                                    Date of Birth:
                        </span>
            </p>
            <p
                        className="other-info"
            >
                        <span>
                                    Birth Place:
                        </span>
            </p>
            <p
                        className="other-info"
            >
                        <span>
                                    Club:
                        </span>
            </p>
            <p
                        className="other-info"
            >
                        <span>
                                    Position:
                        </span>
            </p>
</div>,
        ],
        "className": "player-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1
                className="player-name"
/>,
              <img
                className="player-image"
                src={undefined}
/>,
            ],
            "className": "name-image-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "player-name",
              },
              "ref": null,
              "rendered": null,
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "player-image",
                "src": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p
                className="other-info"
>
                <span>
                                Nationality:
                </span>
</p>,
              <p
                className="other-info"
>
                <span>
                                Date of Birth:
                </span>
</p>,
              <p
                className="other-info"
>
                <span>
                                Birth Place:
                </span>
</p>,
              <p
                className="other-info"
>
                <span>
                                Club:
                </span>
</p>,
              <p
                className="other-info"
>
                <span>
                                Position:
                </span>
</p>,
            ],
            "className": "other-info-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Nationality:
</span>,
                  undefined,
                ],
                "className": "other-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nationality:",
                  },
                  "ref": null,
                  "rendered": "Nationality:",
                  "type": "span",
                },
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Date of Birth:
</span>,
                  undefined,
                ],
                "className": "other-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date of Birth:",
                  },
                  "ref": null,
                  "rendered": "Date of Birth:",
                  "type": "span",
                },
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Birth Place:
</span>,
                  undefined,
                ],
                "className": "other-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Birth Place:",
                  },
                  "ref": null,
                  "rendered": "Birth Place:",
                  "type": "span",
                },
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Club:
</span>,
                  undefined,
                ],
                "className": "other-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Club:",
                  },
                  "ref": null,
                  "rendered": "Club:",
                  "type": "span",
                },
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    Position:
</span>,
                  undefined,
                ],
                "className": "other-info",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Position:",
                  },
                  "ref": null,
                  "rendered": "Position:",
                  "type": "span",
                },
                undefined,
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="player-container"
>
          <div
                    className="name-image-container"
          >
                    <h1
                              className="player-name"
                    />
                    <img
                              className="player-image"
                              src={undefined}
                    />
          </div>
          <div
                    className="other-info-container"
          >
                    <p
                              className="other-info"
                    >
                              <span>
                                        Nationality:
                              </span>
                    </p>
                    <p
                              className="other-info"
                    >
                              <span>
                                        Date of Birth:
                              </span>
                    </p>
                    <p
                              className="other-info"
                    >
                              <span>
                                        Birth Place:
                              </span>
                    </p>
                    <p
                              className="other-info"
                    >
                              <span>
                                        Club:
                              </span>
                    </p>
                    <p
                              className="other-info"
                    >
                              <span>
                                        Position:
                              </span>
                    </p>
          </div>
</div>,
        "className": "container-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="name-image-container"
>
              <h1
                            className="player-name"
              />
              <img
                            className="player-image"
                            src={undefined}
              />
</div>,
            <div
              className="other-info-container"
>
              <p
                            className="other-info"
              >
                            <span>
                                          Nationality:
                            </span>
              </p>
              <p
                            className="other-info"
              >
                            <span>
                                          Date of Birth:
                            </span>
              </p>
              <p
                            className="other-info"
              >
                            <span>
                                          Birth Place:
                            </span>
              </p>
              <p
                            className="other-info"
              >
                            <span>
                                          Club:
                            </span>
              </p>
              <p
                            className="other-info"
              >
                            <span>
                                          Position:
                            </span>
              </p>
</div>,
          ],
          "className": "player-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1
                  className="player-name"
/>,
                <img
                  className="player-image"
                  src={undefined}
/>,
              ],
              "className": "name-image-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                  "className": "player-name",
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "player-image",
                  "src": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="other-info"
>
                  <span>
                                    Nationality:
                  </span>
</p>,
                <p
                  className="other-info"
>
                  <span>
                                    Date of Birth:
                  </span>
</p>,
                <p
                  className="other-info"
>
                  <span>
                                    Birth Place:
                  </span>
</p>,
                <p
                  className="other-info"
>
                  <span>
                                    Club:
                  </span>
</p>,
                <p
                  className="other-info"
>
                  <span>
                                    Position:
                  </span>
</p>,
              ],
              "className": "other-info-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Nationality:
</span>,
                    undefined,
                  ],
                  "className": "other-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Nationality:",
                    },
                    "ref": null,
                    "rendered": "Nationality:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Date of Birth:
</span>,
                    undefined,
                  ],
                  "className": "other-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date of Birth:",
                    },
                    "ref": null,
                    "rendered": "Date of Birth:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Birth Place:
</span>,
                    undefined,
                  ],
                  "className": "other-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Birth Place:",
                    },
                    "ref": null,
                    "rendered": "Birth Place:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Club:
</span>,
                    undefined,
                  ],
                  "className": "other-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Club:",
                    },
                    "ref": null,
                    "rendered": "Club:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      Position:
</span>,
                    undefined,
                  ],
                  "className": "other-info",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Position:",
                    },
                    "ref": null,
                    "rendered": "Position:",
                    "type": "span",
                  },
                  undefined,
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
